content/source/docs/extend/resources.html.md

```diff
@@ -0,0 +1,368 @@
+            client.ExampleInstaceStateCreating,
+            client.ExampleInstaceStateVerifying,
+        },
+        Target: []string{
+            client.ExampleInstaceStateCreateComplete,
+        },
+        Refresh: func() (interface{}, string, error) {
+            resp, err := client.DescribeInstance(name)
+            if err != nil {
+                0, "", err
+            }
+            return resp, resp.Status, nil
+        },
+        Timeout:    d.Timeout(schema.TimeoutCreate),
+        Delay:      10 * time.Second,
+        MinTimeout: 5 * time.Second,
+        ContinuousTargetOccurence: 5,
+    }
+    _, err = createStateConf.WaitForState()
+    if err != nil {
+        return fmt.Errorf("Error waiting for example instance (%s) to be created: %s", d.Id(), err)
+    }
+
+    return resourceExampleInstanceRead(d, meta)
+}
+```
+
+## Importers
+
+Many users migrating to Terraform often have manually managed infrastructure they want to bring under the management of Terraform. Terraform provides a mechanism known as an importer to consolidate those resources into state. As of writing the user will still have to write configuration that will be associated to the import.
+
+When importing the user will specify the configuration address and id of the resource
+
+```
+terraform import example_instance.foo 000-0000
+```
+
+A resources `READ` function will perform a lookup based on the configured `id`. To support this Terraform provides a convenience that allows this passthrough.
+
+```go
+package example
+
+import (
+    "fmt"
+
+    "github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceExampleInstance() *schema.Resource {
+    return &schema.Resource{
+        Create: resourceExampleInstanceCreate,
+        Read:   resourceExampleInstanceRead,
+        Update: resourceExampleInstanceUpdate,
+        Delete: resourceExampleInstanceDelete,
+
+        Schema: map[string]*schema.Schema{
+            "name": {
+                Type:     schema.TypeString,
+                Required: true,
+            },
+        },
+        Importer: &schema.ResourceImporter{
+            State: schema.ImportStatePassthrough,
+        },
+    }
+}
+
+func resourceExampleInstanceRead(d *schema.ResourceData, meta interface{}) error {
+    client := m.(*MyClient)
+
+    obj, err := client.Get(d.Id())
+
+    if err != nil {
+        d.SetId("")
+        return fmt.Errorf("Error retrieving example instance: %s: %s", d.Id(), err)
+    }
+
+    d.Set("name", obj.Name)
+    return nil
+}
+```
+
+There are other cases where the `READ` function uses configuration parameters as the identifier or support for importing multiple resources, as seen in the [AWS Provider](https://github.com/terraform-providers/terraform-provider-aws/blob/d3fe7e9907263b1aa41ddc0736a34b42899d1536/aws/import_aws_dx_gateway.go#L12) is needed. In general it is advised to stick to the passthrough importer when possible.
+
+## Customizing Differences
```

Comment:
Not necessarily a blocker, but are we going to leave out `DiffSuppressFunc` here? I feel like it's even more commonly used than `CustomizeDiff` and probably deserves a few sentences?

Reply:
DiffSuppressFunc is covered in schemas:
https://www.terraform.io/docs/extend/schemas/schema-behaviors.html#diffsuppressfunc

This doc focuses on Resource APIs, as far as I know DiffSuppressFunc is at the schema level?

Reply:
Ah, doh, you're right, I missed that.